optaplanner-core/src/main/java/org/optaplanner/core/impl/io/jaxb/GenericJaxbIO.java

```diff
@@ -0,0 +1,318 @@
+                rootClass.getName(), Arrays.toString(elementNamespaceOverrides));
+
+        // Create a SAXParser to use its XMLReader on the XMLFilter
+        SAXParserFactory saxParserFactory = SAXParserFactory.newInstance();
+        SAXParser saxParser;
+        try {
+            // Protect the parser against the XXE attack
+            // https://owasp.org/www-project-top-ten/OWASP_Top_Ten_2017/Top_10-2017_A4-XML_External_Entities_(XXE)
+            saxParserFactory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            saxParser = saxParserFactory.newSAXParser();
+        } catch (ParserConfigurationException | SAXException e) {
+            throw new OptaPlannerXmlSerializationException(errMessage, e);
+        }
+        XMLReader xmlReader;
+        try {
+            xmlReader = saxParser.getXMLReader();
+        } catch (SAXException e) {
+            throw new OptaPlannerXmlSerializationException(errMessage, e);
+        }
+
+        XMLFilter namespaceOverridingXmlFilter = new NamespaceOverridingXmlFilter(xmlReader, elementNamespaceOverrides);
+        namespaceOverridingXmlFilter.setParent(xmlReader);
+
+        // Use UnmarshallerHandler as a content handler for the XML filter.
+        Unmarshaller unmarshaller = createUnmarshaller();
+        UnmarshallerHandler unmarshallerHandler = unmarshaller.getUnmarshallerHandler();
+        namespaceOverridingXmlFilter.setContentHandler(unmarshallerHandler);
+
+        InputSource xmlInputSource = new InputSource(reader);
+        try {
+            // Parse the XML to feed its content into the UnmarshallerHandler.
+            namespaceOverridingXmlFilter.parse(xmlInputSource);
+        } catch (IOException | SAXException e) {
+            throw new OptaPlannerXmlSerializationException(errMessage, e);
+        }
+
+        try {
+            return (T) unmarshallerHandler.getResult();
+        } catch (JAXBException e) {
+            throw new OptaPlannerXmlSerializationException(errMessage, e);
+        }
+    }
+
+    @Override
+    public void write(T root, Writer writer) {
+        Objects.requireNonNull(root);
+        Objects.requireNonNull(writer);
+        DOMResult domResult = new DOMResult();
+        try {
+            marshaller.marshal(root, domResult);
+        } catch (JAXBException jaxbException) {
+            String errMessage = String.format(ERR_MSG_WRITE, rootClass.getName());
+            throw new OptaPlannerXmlSerializationException(errMessage, jaxbException);
+        }
+
+        formatXml(new DOMSource(domResult.getNode()), null, writer);
+    }
+
+    public void writeWithoutNamespaces(T root, Writer writer) {
+        Objects.requireNonNull(root);
+        Objects.requireNonNull(writer);
+        DOMResult domResult = new DOMResult();
+        final String errMessage = String.format(ERR_MSG_WRITE, rootClass.getName());
```

Comment:
The errorMessage String should only be created if it actually fails, so inside the catch